xfr.go

```diff
@@ -198,11 +198,16 @@ func (t *Transfer) Out(w ResponseWriter, q *Msg, ch chan *Envelope) error {
 		r.Authoritative = true
 		// assume it fits TODO(miek): fix
 		r.Answer = append(r.Answer, x.RR...)
+		if tsig := q.IsTsig(); tsig != nil {
```

Comment:
Can you merge these two `if`-statements? So: `tsig != nil && w.TsigStatus() == nil`.